src/config.ts

```diff
@@ -51,53 +52,89 @@ export default class ConfigService {
    */
   #currentVersion: string | null = null;
 
-  constructor(path: string) {
-    this.#uri = path;
+  constructor(uri: string) {
+    this.#uri = uri;
   }
 
-  async resolve() {
-    let url: URL | null = null;
+  /**
+   * Fetch and return the configuration module.
+   *
+   * @param deadline When a deadline is provided, fetching the module via HTTP
+   *                 will be aborted if the deadline is exceeded. Specified in
+   *                 milliseconds.
+   */
+  async fetchConfig(deadline?: number) {
+    if (!this.configIsViaHttp) {
+      return require(/* webpackIgnore: true */ this.#uri).default;
+    }
 
-    try {
-      url = new URL(this.#uri);
-    } catch {
-      // Not a valid URL
+    // Handle fetch deadline via an AbortController
+    const ac = new AbortController();
+    let deadlineTimeout: NodeJS.Timeout | null = null;
+    let config: any = null;
+
+    if (deadline !== undefined) {
+      deadlineTimeout = setTimeout(() => ac.abort(), deadline);
     }
 
-    const isHttpUrl = url?.protocol?.startsWith('http');
-    logger.info(`Resolving render config via ${isHttpUrl ? 'HTTP' : 'provided file'}`);
+    config = await loadViaHttp(this.#uri, ac);
+
+    if (deadlineTimeout !== null) {
+      clearTimeout(deadlineTimeout);
+    }
+
+    return config;
+  }
 
-    const config = isHttpUrl
-      ? await loadViaHttp(this.#uri)
-      : require(/* webpackIgnore: true */ this.#uri).default;
+  /**
+   * Resolve the provided configuration URI. If the configuration cannot be
+   * resolved, an error will be thrown.
+   */
+  async resolveEnsured() {
```

Comment:
Does this need to be a separate function?  It's basically the same as if you did the `resolveWithPolling` with an option for `onlyPollOnce` since it is immediate on the first call anyway.

Reply:
I don't think it _needs_ to, but I kind of prefer to keep the implementation a bit separate since it also throws errors, and I wouldn't mind having the ConfigPoller just be responsible for polling.